src/TreeView/TreeView.js

```diff
@@ -0,0 +1,471 @@
+    isExpanded: PropTypes.bool,
+    level: PropTypes.number,
+    onExpandClick: PropTypes.func
+};
+
+TreeItem.defaultProps = {
+    expandData: {},
+    level: 0,
+    onExpandClick: () => {}
+};
+
+TreeItem.propDescriptions = {
+    expandData: '_INTERNAL USE ONLY._',
+    isExpanded: 'Set to *true* for expanded tree item. This variable is handled internally, but can be overridden by the consumer through this prop',
+    level: '_INTERNAL USE ONLY._',
+    onExpandClick: '_INTERNAL USE ONLY._'
+};
+
+export class TreeBranch extends Component {
+    render() {
+        return <Tree {...this.props} />;
+    }
+}
+
+TreeBranch.displayName = 'TreeBranch';
+
+TreeBranch.propTypes = {
+    children: PropTypes.node,
+    expandData: PropTypes.object,
+    isExpanded: PropTypes.bool,
+    level: PropTypes.number,
+    onExpandClick: PropTypes.func
+};
+
+TreeBranch.defaultProps = {
+    expandData: {},
+    level: 0
+};
+
+TreeBranch.propDescriptions = {
+    expandData: '_INTERNAL USE ONLY._',
+    isExpanded: '_INTERNAL USE ONLY._',
+    level: '_INTERNAL USE ONLY._',
+    onExpandClick: '_INTERNAL USE ONLY._'
+};
+
+export class Tree extends Component {
+    render() {
+        const {
+            children,
+            expandData,
+            level,
+            isExpanded,
+            onExpandClick,
+            ...rest
+        } = this.props;
+
+        const className = classnames({
+            'fd-tree': level === 0,
+            'fd-tree__group': level > 0,
+            [`fd-tree__group--sublevel-${level}`]: level > 0,
+            'is-hidden': level > 0 && !isExpanded
+        });
+
+        return (
+            <ul
+                {...rest}
+                aria-hidden={level > 0 && !isExpanded}
+                className={className}
+                role={level === 0 ? 'tree' : 'group'}>
+                {
+                    React.Children.map(children, (child) => {
+                        return React.cloneElement(child, {
+                            expandData,
+                            level,
+                            onExpandClick
+                        });
+                    })
+                }
+            </ul>
+        );
+    }
+}
+
+Tree.displayName = 'Tree';
+
+Tree.propTypes = {
+    children: PropTypes.node,
+    expandData: PropTypes.object,
+    isExpanded: PropTypes.bool,
+    level: PropTypes.number,
+    onExpandClick: PropTypes.func
+};
+
+Tree.defaultProps = {
+    expandData: {},
+    level: 0
+};
+
+Tree.propDescriptions = {
+    expandData: '_INTERNAL USE ONLY._',
+    isExpanded: '_INTERNAL USE ONLY._',
+    level: '_INTERNAL USE ONLY._',
+    onExpandClick: '_INTERNAL USE ONLY._'
+};
+
+export class TreeView extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            expandData: {},
+            isExpandAll: false
+        };
+    }
+
+    static getDerivedStateFromProps(props, state) {
+        const {
+            isExpandAll
```

Comment:
It seems like this should also accept `expandData` as a prop so the user can control which items may be initially expanded.  That would also mean that prop would be checked in `getDerivedStateFromProps `.  If that gets added, then I think there are a few more things that would need to be added:

* Add a callback prop to `TreeView` that gets called whenever the expand/collapse state changes on any row id.
* Add a `rowId` prop to `TreeItem` so a consumer could set their own id value (since they are the keys in the `expandData` object).

Reply:
Exposed `expandData` props and callbacks in 98d4612.

Ex API:
```
<TreeView expandData={{myRowId: true}}>
    <Tree>
         <TreeItem rowId={'myRowId'}>
                 <TreeRow>
                      <TreeCol>Row 1</TreeCol>
                   </TreeRow>
           </TreeItem>
    </Tree>
</TreeView>
```